Add ability to use `workspaceFolder` in Visual studio and VSCode to interpolate settings (#891)
microsoft/cadl: packages/cadl-vscode/README.md

See https://github.com/microsoft/cadl#installing-vs-code-extension for installation instructions.

**NOTE**: The npm package is used as an implementation detail of the `cadl vscode` command that installs the Visual Studio Cadl extension, and not intended to be used for other purposes.

## Configure VSCode extension

Cadl wil interpolate a few variables using this pattern `${<name>}`. For example `${workspaceFolder}`.

Available variables:

- `workspaceFolder`: Corespond to the root of your Visual Studio workspace.

### `cadl.cadl-server.path`: Configure the server path

> There is cases where the Cadl project is located in a subfolder. In this case Cadl extension is not able to find the cadl compiler automatically and needs a little guidance.

Reviewer: ```suggestion
There are cases where the Cadl project is located in a subfolder. In such cases, the Cadl extension is not able to find the cadl compiler automatically and needs a little guidance.
```

> This setting provide the ability to configure where the cadl compiler is located.

Reviewer: ```suggestion
This setting provides the ability to configure where the cadl compiler is located.
```
